lib/ansible/vars/plugins.py

```diff
@@ -36,6 +37,9 @@ def get_plugin_vars(loader, plugin, path, entities):
                 raise AnsibleError("Cannot use v1 type vars plugin %s from %s" % (plugin._load_name, plugin._original_path))
             else:
                 raise AnsibleError("Invalid vars plugin %s from %s" % (plugin._load_name, plugin._original_path))
+
+    validate_variable_names(data.keys())
```

Comment:
redundant since you already check from vars manager when taking the variables from the plugin

Reply:
So this appears to be actually the right place. The previous `_get_plugin_vars` in vars manager was unused. So I put this back and removed the unused method in https://github.com/ansible/ansible/pull/58062/commits/6cd6dc03610bce6e7d284fb628ce19c09d562c08 (in this PR).